### PR TITLE
[Driver] Clean up unused architecture related bits for *BSD's

### PR DESCRIPTION
**Pull request by brad0**
- FreeBSD removed big-endian arm with 12.0.
- OpenBSD never had big-endian arm support. I added it just in case, but it has never been used.
- Remove sparcel bits. It was sprinkled in a few places but it will never be a thing.
- Remove 32-bit sparc bits for FreeBSD. FreeBSD has never had 32-bit sparc support.
- Remove sparc64 IAS test as support was enabled across the board awhile ago.

### CODE REVIEW

**Comment by brad0** [comment]
Fix a typo in the commit message.

**Comment by brad0** [comment]
ping.

File: clang/lib/Basic/Targets.cpp [hunk 1]
```diff
@@ -247,12 +247,8 @@ std::unique_ptr<TargetInfo> AllocateTarget(const llvm::Triple &Triple,
     switch (os) {
     case llvm::Triple::Linux:
       return std::make_unique<LinuxTargetInfo<ARMbeTargetInfo>>(Triple, Opts);
-    case llvm::Triple::FreeBSD:
```
Comment (emaste) [inline clang/lib/Basic/Targets.cpp:250]: OK - there are no supported FreeBSD releases left that support armeb, and it was uncommon enough that I can't imagine anyone wants to use contemporary clang to target unsupported releases.

File: clang/test/Driver/freebsd.c [hunk 2]
```diff
@@ -163,9 +163,9 @@
 // CHECK-ARM-EABIHF-NOT: as{{.*}}" "-mfpu=softvfp"
 // CHECK-ARM-EABIHF-NOT: as{{.*}}" "-matpcs"
 
-// RUN: %clang --target=sparc-unknown-freebsd -### %s -fpic -no-integrated-as 2>&1 \
-// RUN:   | FileCheck --check-prefix=CHECK-SPARC-PIE %s
-// CHECK-SPARC-PIE: as{{.*}}" "-KPIC
+// RUN: %clang --target=sparc64-unknown-freebsd -### %s -fpic -no-integrated-as 2>&1 \
```
Comment (emaste) [inline clang/test/Driver/freebsd.c:166]: fwiw sparc64 is still supported in FreeBSD 12.x, but that is EOL at the end of the year, so this could go before too long as well

File: clang/test/Driver/freebsd.c [hunk 3]
```diff
@@ -191,11 +191,6 @@
 // RUN:   | FileCheck -check-prefix=CHECK-MIPS64-CPU %s
 // CHECK-MIPS64-CPU: "-target-cpu" "mips3"
 
-// Check that the integrated assembler is enabled for SPARC64
```
Comment (emaste) [inline clang/test/Driver/freebsd.c:194]: why is this part here (before sparc64 is removed from FreeBSD)? It is not covered by the commit message anyway, afaict

**Review by emaste — COMMENTED** [review]
overall looks fine, no objection from FreeBSD.
one open question.

**Comment by MaskRay** [comment]
Thanks for the cleanup!

File: clang/test/Driver/freebsd.c (hunk 2 above)
Comment (brad0) [reply clang/test/Driver/freebsd.c:166]: Ya, I didn't plan on removing the sparc64 support.

File: clang/test/Driver/freebsd.c (hunk 3 above)
Comment (brad0) [reply clang/test/Driver/freebsd.c:194]: I already removed the other OS specific tests as I enabled IAS across the board awhile ago.

Comment (emaste) [reply clang/test/Driver/freebsd.c:194]: Maybe worth noting that in the commit message too but either way LGTM for FreeBSD

**Review by emaste — APPROVED** [review]

File: clang/lib/Basic/Targets.cpp (hunk 1 above)
Comment (brad0) [reply clang/lib/Basic/Targets.cpp:250]: Ya, even with support being removed I didn't think it was common at all.

File: clang/test/Driver/freebsd.c (hunk 3 above)
Comment (brad0) [reply clang/test/Driver/freebsd.c:194]: Done.
